### ISSUE TITLE
Implement a plant-disease image classifier behind the existing seam

### ISSUE THREAD

**Issue by Rekin226** [Rekin226/group1--project#71]
A general vision model is mediocre at fine-grained leaf disease. A specialist classifier is much better at it. `agent/classifier.py` is a **finished seam with no backend** — implementing one is a high-impact, self-contained contribution.

### Why it's wired the unusual way

The obvious design is to let the classifier's label be the answer. We deliberately don't, because a label like `Tomato___Late_blight` is a diagnosis with **no citation, no discriminating check, and no statement of what it cannot see** — everything this project exists to avoid.

Instead the classifier is a third source of *categorical features*:

```
photo ──▶ VLM ───────────▶ prose ──▶ features ─┐
    └──▶ classifier ──▶ label ──▶ features ────┴─▶ aqua_model.triage ──▶ cited differential
```

The label maps, through a fixed table, onto the **visible evidence a diagnosis would have rested on** — `late blight` becomes "brown spots on leaves", never "late blight". Consequences:

- a classifier can never introduce an uncited candidate (there's a test named exactly that);
- a wrong label can only pick a different branch of a differential, never inject a number;
- the raw label is still shown to the user, clearly marked unverified and explicitly *outside* the cited differential.

### The task

1. Register a default model in `DEFAULT_MODELS` and build it in `_build_classifier_backend`. **Import your library inside the function** — `agent/vision.py` and `agent/transcribe.py` are the pattern — so it never becomes a hard dependency for people who don't use it.
2. Return `[Prediction(label, confidence)]`, or `[(label, confidence)]` tuples; `make_classifier` normalises either.
3. Map any label vocabulary your model uses in `_LABEL_FEATURES`, to visible evidence only.
4. Tests with a stub backend — no model download in CI. `agent/tests/test_classifier.py` shows the shape.

### Model suggestions

- **PlantVillage-trained models** are the obvious start — 38 classes, openly licensed, many pretrained checkpoints on Hugging Face. Big caveat worth stating in your PR: PlantVillage is lab-style single leaves on uniform backgrounds, and it degrades hard on real field photos. That's honest and fine; the confidence threshold (`MIN_CONFIDENCE = 0.60`) exists for this.
- A **fish disease** classifier would be even more valuable and barely exists publicly. If you know of a dataset, an issue pointing at it is itself a contribution.
- Inference should be **CPU-viable**. The deployment target is a modest VPS or a laptop, not a GPU box. ONNX Runtime or a small torchvision model is more in keeping than anything large.

### What would make a PR land fast

- The label→features mapping is the part that needs care. Ask yourself for each class: *what would a person actually SEE?* — and map that.
- Say plainly in the PR what your model is bad at. That's a feature here, not a weakness.
- If your classifier reports an unknown class, it should map to nothing rather than guess.

### Related

- The Tier-2 photo corpus that would let us measure any of this is still empty — see the photo-donation issue. A classifier and a corpus make each other much more valuable.
- Read [CONTRIBUTING.md § Adding an image classifier](https://github.com/Rekin226/Agronaut/blob/main/CONTRIBUTING.md#adding-an-image-classifier) first.

**Comment by kr1xh** [comment]
I'd like to take #71.

I’ve looked through `agent/classifier.py`, the existing classifier tests, and the contributing guidelines. I’ll implement a CPU-viable PlantVillage-style backend behind the existing `_build_classifier_backend()` seam, keeping the dependency lazy and optional.

The implementation will:

* register the backend/model in `DEFAULT_MODELS`
* return `Prediction(label, confidence)` through the existing interface
* reuse the current `_LABEL_FEATURES` approach so classifier output contributes only visible evidence, never a diagnosis
* keep model downloads out of CI and use the existing stub-based test pattern
* handle unknown labels conservatively by mapping them to no additional features
* document the chosen model/checkpoint and its limitations, especially the gap between PlantVillage-style images and real field photos

I’ll first verify the specific checkpoint’s licence, label vocabulary, and CPU inference requirements before wiring it in.

Could you assign #71 to me?

**Comment by Rekin226** [comment]
@kr1xh It's yours. Your read of the seam is right — lazy import inside `_build_classifier_backend`, `Prediction(label, confidence)` out, `_LABEL_FEATURES` mapping to visible evidence only, stub tests with no download in CI. That's the whole contract, and naming "unknown labels map to no additional features" unprompted tells me you read the tests rather than the summary.

One thing to decide early, because it shapes the PR: PlantVillage's 38 classes are tomato/potato/apple-heavy, and aquaponics growers mostly run lettuce, basil and leafy greens. Say in the PR which of your classes can actually fire for our users. A classifier honest about covering four of our crops is worth more than one claiming 38 that silently never match.

#93 is merged and the tree is clean under ruff, so nothing's blocking you.
